Tests/eppo/RuleEvaluatorTests.swift

```diff
@@ -2,195 +2,257 @@ import XCTest
+            subjectAttributes: SubjectAttributes()
+        );
+        XCTAssertEqual(flagEvaluation.flagKey, "test");
+        XCTAssertEqual(flagEvaluation.subjectKey, "test");
+        //XCTAssertEqual(flagEvaluation.subjectAttributes, SubjectAttributes());
+        XCTAssertNil(flagEvaluation.allocationKey);
+        XCTAssertNil(flagEvaluation.variation);
+        XCTAssertEqual(flagEvaluation.extraLogging, [:]);
+        XCTAssertFalse(flagEvaluation.doLog);
+    }
+}
+
+final class flagEvaluatorTests: XCTestCase {
+    var flagEvaluator: FlagEvaluator!
+
+    let flag = UFC_Flag(
+        key: "test",
+        enabled: true,
+        variationType: UFC_VariationType.string,
+        variations: [:],
+        allocations: [],
+        totalShards: 100
+    );
+    
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        let lookup = ["subject_key": 50]
+        let sharder = DeterministicSharder(lookup: lookup)
+        flagEvaluator = FlagEvaluator(sharder: sharder)
+    }
+
+    public func testDisabledFlag() {
+        let flag = createFlag(flag: flag, rules: [], enabled: false);
+        let flagEvaluation = flagEvaluator.evaluateFlag(flag: flag, subjectKey: "subject_key", subjectAttributes: SubjectAttributes());
+
+        XCTAssertFalse(flagEvaluation.doLog)
+    }
+
     public func testMatchesAnyRuleWithEmptyConditions() throws {
-        let targetingRuleWithEmptyConditions: TargetingRule = self.createRule([]);
-        let targetingRules: [TargetingRule] = [targetingRuleWithEmptyConditions];
+        let targetingRuleWithEmptyConditions: UFC_Rule = UFC_Rule(conditions: []);
+        let targetingRules: [UFC_Rule] = [targetingRuleWithEmptyConditions];
+        let testFlag = createFlag(flag: flag, rules: targetingRules);
         
         var subjectAttributes: SubjectAttributes = SubjectAttributes();
         subjectAttributes["name"] = EppoValue.valueOf("test");
 
-        XCTAssertEqual(
-            targetingRuleWithEmptyConditions,
-            try RuleEvaluator.findMatchingRule(subjectAttributes, targetingRules)
-        );
+        let evaluationResult = flagEvaluator.evaluateFlag(flag: testFlag, subjectKey: "subject_key", subjectAttributes: subjectAttributes)
+        XCTAssertTrue(evaluationResult.doLog)
     }
     
     public func testMatchesAnyRuleWithEmptyRules() throws {
-        let targetingRules: [TargetingRule] = [];
+        let targetingRules: [UFC_Rule] = [];
+        let testFlag = createFlag(flag: flag, rules: targetingRules);
+
         var subjectAttributes: SubjectAttributes = SubjectAttributes();
         subjectAttributes["name"] = EppoValue.valueOf("test");
         
-        XCTAssertNil(try RuleEvaluator.findMatchingRule(subjectAttributes, targetingRules));
+        let evaluationResult = flagEvaluator.evaluateFlag(flag: testFlag, subjectKey: "subject_key", subjectAttributes: subjectAttributes)
+        XCTAssertTrue(evaluationResult.doLog)
```

Comment:
While we don't need to assert every field of the evaluation result since we cover that with the first test, I think for all these other tests we'll still want to assert `variation` in addition to `do Log`.